### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by AntonioGomez12**
websitelink: https://icy-ocean-04e12481e.3.azurestaticapps.net

### CODE REVIEW

**Comment by NicholasBurlakov** [comment]
Make all pages responsive ✅
1080p desktop ✅
iPad Air ✅ 
Samsung Galaxy S20 Ultra Super Mega Excellent ✅ 
IPhone SE ✅
Create a improved landing page and site theme (Background, make it look attractive to play) ✅
Theme should extend all pages ✅
Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ❌
Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌
Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌
It should indicate if the user has played that day ❌
Clicking on the item should take the player to that game (Extra Credit) ❌

In terms of responsiveness i didnt check if it works on ipad or samsung (becouse i dont own them) but if you took Grants code then he made them responsive. I went out on a limb and assumed that it works.

**Comment by roydunn19** [comment]
missing instructions page

**Comment by roydunn19** [comment]
missing word of the day page

**Comment by roydunn19** [comment]
page isnt responsive. to check this inspect the page and check if it is resposive on iphone se which is the smallest frame. things i noticed are dialogs go off screen, landing page title is clipped off screen.

**Comment by roydunn19** [comment]
i would recommend using overflow: hidden to fix your title having a ellipse when shrunk to iphone size

File: wordle-web/src/views/WordleView.vue [hunk 1]
```diff
@@ -1,118 +1,127 @@
...
-  <v-overlay :model-value="overlay" class="align-center justify-center" persistent>
-    <v-progress-circular color="primary" indeterminate size="64" />
-  </v-overlay>
+  <main class="px-4 mt-4">
+    <v-overlay :model-value="overlay" class="align-center justify-center" persistent>
+      <v-progress-circular color="primary" indeterminate size="64" />
+    </v-overlay>
 
-  <div class="text-h4 text-center">Wordle Mind Bender</div>
+    <h1 class="text-center">Timer: {{ timer }}</h1>
```
Comment (roydunn19) [inline wordle-web/src/views/WordleView.vue:7]: this is a cool addition to the game

File: wordle-web/src/router/index.ts
Comment (roydunn19) [inline wordle-web/src/router/index.ts:1]: missing view for wordle of the day

**Comment by roydunn19** [comment]
nice logo on landing page bit its missing the theme that extends across all pages

File: Wordle.Api/Wordle.Api/Services/PlayerService.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+{
+    public class PlayerService
+    {
+        private readonly AppDbContext _db;
+        public PlayerService(AppDbContext db)
+        {
+            _db = db;
+        }
+
+        public async Task<IEnumerable<Player>> GetTopTen()
```
Comment (mmwoodfo) [inline Wordle.Api/Wordle.Api/Services/PlayerService.cs:14]: Alot of the changes from this file were from Assignment 3. Make sure to only include changes required for Assignment 4.

File: wordle-web/src/App.vue [hunk 3]
```diff
@@ -1,47 +1,107 @@
...
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+
   <v-app>
     <v-app-bar :elevation="3">
       <template v-slot>
         <v-app-bar-title>
-          <RouterLink to="/">
+          <RouterLink to="/game">
```
Comment (mmwoodfo) [inline wordle-web/src/App.vue:18]: This should stay linking to the home page

File: wordle-web/src/views/WordleOfTheDay.vue [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  const username = $cookies?.get('username')
+  Axios.post('https://wurdle.azurewebsites.net/Player/AddPlayerFromBody', {
+    name: username,
+    numberOfAttempts: game.guesses.indexOf(game.guess),
+    elapsedSeconds: timer.value
+  }).catch((error) => {
+    console.log(error)
+  })
+}
+function newGame() {
```
Comment (mmwoodfo) [inline wordle-web/src/views/WordleOfTheDay.vue:79]: This should get you the same word for the current day.

File: wordle-web/src/views/LeaderBoardView.vue [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      </v-table>
+    </div>
+  </main>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue'
+import Axios from 'axios'
+interface LeaderboardData {
+  playerId: number
```
Comment (mmwoodfo) [inline wordle-web/src/views/LeaderBoardView.vue:40]: It doesn't look like you have a leaderboard for the recent words? Just the leaderboard from Assignment 3 for the players

**Review by mmwoodfo — COMMENTED** [review]
Looks like you have a couple of issues with different sized screens
image

**Pull request closed without merging**